### PR TITLE
chore: api testing test nits

### PR DESCRIPTION
**Pull request by mxschmitt**

### CODE REVIEW

File: packages/playwright-core/src/server/fetch.ts [hunk 1]
```diff
@@ -155,7 +155,7 @@ export abstract class APIRequestContext extends SdkObject {
         return { error: `${fetchResponse.status} ${fetchResponse.statusText}` };
       return { fetchResponse: { ...fetchResponse, fetchUid } };
     } catch (e) {
-      return { error: String(e) };
+      return { error: e instanceof Error ? e.message : String(e) };
```
Comment (mxschmitt) [inline packages/playwright-core/src/server/fetch.ts:158]: this fixes an issue that the actual error got wrapped as `Error: actual error` and landed by that not in the ideal form in the language ports.

File: tests/page/page-network-response.spec.ts [hunk 2]
```diff
@@ -267,3 +268,12 @@ it('should behave the same way for headers and allHeaders', async ({ page, serve
   expect(allHeaders['name-b']).toEqual('v4');
 });
 
+it('should should provide a Response with a file URL', async ({ page, asset, isAndroid, browserName }) => {
```
Comment (yury-s) [inline tests/page/page-network-response.spec.ts:271]: typo: should should

File: tests/global-fetch.spec.ts [hunk 3]
```diff
@@ -41,27 +42,26 @@ it.afterAll(() => {
 for (const method of ['fetch', 'delete', 'get', 'head', 'patch', 'post', 'put']) {
   it(`${method} should work`, async ({ playwright, server }) => {
     const request = await playwright.request.newContext();
-    const response = await request[method](server.PREFIX + '/simple.json');
+    const response: APIResponse = await request[method](server.PREFIX + '/simple.json');
```
Comment (yury-s) [inline tests/global-fetch.spec.ts:45]: why do we need this? isn't tsc smart enough?

File: packages/playwright-core/src/server/fetch.ts (hunk 1 above)
Comment (yury-s) [reply packages/playwright-core/src/server/fetch.ts:158]: I believe we are losing backand's stack trace in this case and it often provides useful information about what's going on (at the very least for us).

File: packages/playwright-core/src/client/fetch.ts [hunk 4]
```diff
@@ -214,7 +214,7 @@ export class APIResponse implements api.APIResponse {
   }
 
   ok(): boolean {
-    return this._initializer.status === 0 || (this._initializer.status >= 200 && this._initializer.status <= 299);
+    return (this._initializer.status >= 200 && this._initializer.status <= 299);
```
Comment (yury-s) [inline packages/playwright-core/src/client/fetch.ts:217]: add a test with file:// url ?

**Review by yury-s — APPROVED** [review]
I'm not sure about the change in error formatting, otherwise looks good.

File: tests/global-fetch.spec.ts (hunk 3 above)
Comment (mxschmitt) [reply tests/global-fetch.spec.ts:45]: good catch. I fixed it the other way around by adding `as const` to the for loop values.

File: packages/playwright-core/src/server/fetch.ts (hunk 1 above)
Comment (mxschmitt) [reply packages/playwright-core/src/server/fetch.ts:158]: Seems like it never added the stack trace with the previous implementation: 
image

File: packages/playwright-core/src/client/fetch.ts (hunk 4 above)
Comment (mxschmitt) [reply packages/playwright-core/src/client/fetch.ts:217]: added one. For our api testing we don't support the 'file:' protocol.

**Pull request merged**